src/workerd/api/streams/standard.c++

```diff
@@ -2905,8 +2858,11 @@ private:
           auto handle = KJ_ASSERT_NONNULL(result.value).getHandle(js);
           if (!handle->IsArrayBufferView() && !handle->IsArrayBuffer()) {
             auto error = js.v8TypeError("This ReadableStream did not return bytes.");
-            doError(js, error);
-            return loop(js);
+            auto rs = kj::mv(readable);
+            state.template init<StreamStates::Errored>(js.v8Ref(error));
```

Comment:
(not a review comment): I wonder if there is a `swap` std function that replaces the value? Maybe in kj? I often see this pattern of `kj:mv`ing things to temp.

Reply:
A swap method on `kj::OneOf`<...>` would be helpful, I think. We've had problems with this kind of pattern in the past where the value was moved but not replaced with something and it caused issues, The follow would be a whole lot nicer/safer

```
auto rs = state.template swap<StreamStates::Errored>(js.v8Ref(error));
```